### ISSUE TITLE
Allow reading of plain text

### ISSUE THREAD

**Issue by sfacets** [cameron/squirt#108]
The bookmarklet currently doesn't allow for this. Even better would be able to read epub and mobi formats

**Comment by vladholubiev** [comment]
I think @sfacets talks about ability to read text from content-editable textboxes. I vote for this feature too.

**Comment by chillingtouch** [comment]
Yeah, please add this feature :8ball:

**Comment by cameron** [comment]
do you mean you can't read a txt file with it? if so, that's because a txt file can't run javascript, and the bookmarklet is javascript. it's possible the content extraction code is overcomplicated and fails to grab body text if that's all there is. 

w/r/t epub and mobi, i don't read those often, so can you tell me how you read them? is it in a browser? in an app?

**Comment by vladholubiev** [comment]
I don't think that somebody often read epub in browser, don't bother with this. Instead you may pay attention to <textarea> blocks. Every time I try to read selected text frim it Squirt says that it is too hard to read. I often face with this while editing documents in Google Drive and articles in Wikipedia.
In the first case it displays the text that my browser doesn;t support Javascript and in second - That this page is too hard to read. I also had problems with this online-notepad, it displays a rubbish such as JS code related to Google Analytics.

**Comment by cameron** [comment]
What browser?

**Comment by chillingtouch** [comment]
Selected text reading doesnt work on FF Nightly 31, under Linux

**Comment by sfacets** [comment]
Wouldn't the JavaScript be run externally from the text file when using the bookmarklet? For instance, if I were to open a plain text document in the browser, the bookmarklet doesn't work.

epub is an open standard, and is basically just a zipped collection of xml/html files
mobi is similar, not sure of how it is put together

They can be read i the browser through a plugin, or on ebook apps such as Calibre, or the Kindle app.

**Comment by vladholubiev** [comment]
@cameron latest 33d Chrome, Win8.1

**Comment by kristoffernolgren** [comment]
A solution for this would be a page on squirt.io that you can paste text into and squirt, this would also work for PDF:s, which is common, specially for students.

**Comment by evanwolf** [comment]
Part of the case for reading ebooks: a gazillion sold/read every day; lots of material only available in book form; they are 10x to 200x longer than most web articles, so readers benefit from RSVP even more; fiction and narrative books (like biographies) are perfect for high speed linear reading. 

About ebook formats. epub, mobi are often read in native app readers. for instance, Amazon's Kindle mobile and desktop apps read .mobi files. Calibre (github: https://github.com/kovidgoyal/calibre) is a popular FOSS desktop ebook client that reads/converts all these ebook files; about 3.5 million active users. 

You already parse html to find the relevant text. For ebooks, I'd want the addition of a few book-specific features: 
- Navigate within the book using table of contents links. ToC is usually provided, along with anchor links.
- Remember last place in book read. Book reading is often interrupted.
- Multiple progress bars: % of section/chapter/book 

Thanks. 
- Phil
